### PR TITLE
chore(fv): integrate annotation tokens into lexer

### PR DESCRIPTION
**Pull request by Aristotelis2002**
Added tokens to the lexer to process our formal verification annotations, which follow the syntax of an attribute. Also added a unit test.

### CODE REVIEW

File: compiler/noirc_frontend/src/lexer/lexer.rs [hunk 1]
```diff
@@ -1363,4 +1363,25 @@ mod tests {
...
+            Token::EOF,
+        ];
+
+        let mut lexer = Lexer::new(input);
+        let token = lexer.next_token().unwrap();
+
+        assert_eq!(
+            token.token(),
+            &Token::Attribute(Attribute::Secondary(crate::token::SecondaryAttribute::Ensures(
+                Tokens(expected_body.iter().map(|x| SpannedToken::new(x.clone(), Span::empty(0))).collect())
```
Comment (smanilov) [inline compiler/noirc_frontend/src/lexer/lexer.rs:1383]: nit: long line, please break into multiple

File: compiler/noirc_frontend/src/lexer/lexer.rs [hunk 2]
```diff
@@ -1363,4 +1363,25 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_ensures_attribute() {
```
Comment (smanilov) [inline compiler/noirc_frontend/src/lexer/lexer.rs:1368]: That's great! Please, add a test for requires as well.

**Review by smanilov — APPROVED** [review]
LGTM, as long as you address the two comments.

File: compiler/noirc_frontend/src/lexer/lexer.rs (hunk 2 above)
Comment (Aristotelis2002) [reply compiler/noirc_frontend/src/lexer/lexer.rs:1368]: Done

File: compiler/noirc_frontend/src/lexer/lexer.rs (hunk 1 above)
Comment (Aristotelis2002) [reply compiler/noirc_frontend/src/lexer/lexer.rs]: Done

**Pull request merged**